ICRCs/ICRC-106/ICRC-106.md

```diff
@@ -0,0 +1,210 @@
+
+These metadata entries allow clients to discover and interact with the index canister associated with a ledger and can be retrieved using method `icrc1_metadata` defined by the ICRC-1 standard.
+
+
+Compliant ledgers MUST also implement the following endpoint for programmatically retrieving the index principal:
+
+```candid
+icrc106_get_index_principal: () -> (principal) query;
+```
+
+The metadata entry `icrc106:index_principal` and the `icrc106_get_index_principal` method MUST provide consistent information. Specifically:
+
+- The `icrc106:index_principal` metadata entry MUST represent the textual form of the principal returned by the `icrc106_get_index_principal` method.  
+- The `icrc106_get_index_principal` method MUST return the principal corresponding to the index canister associated with the ledger, as specified in the `icrc106:index_principal` metadata entry.
+
+This requirement ensures that both mechanisms reliably point to the same index canister. Implementations
+
+
+## 3. Index Canister Interface
+
+The index canister associated with the ledger SHOULD implement the following minimal Candid interface:
+
+```candid
+type Tokens = nat;
+
+type BlockIndex = nat;
+
+type SubAccount = blob;
+
+type Account = record {
+    owner: principal;
+    subaccount: opt SubAccount;
+};
+
+type GetAccountTransactionsArgs = record {
+    account: Account;
+    start: opt BlockIndex; // The block index of the last transaction seen by the client.
+    max_results: nat; // Maximum number of transactions to fetch.
+};
+
+type Burn = record {
+    from : Account;
+    memo : opt vec nat8;
+    created_at_time : opt nat64;
+    amount : nat;
+    spender : opt Account;
+};
+
+type Transfer = record {
+    to : Account;
+    fee : opt nat;
+    from : Account;
+    memo : opt vec nat8;
+    created_at_time : opt nat64;
+    amount : nat;
+    spender : opt Account;
+};
+
+type Approve = record {
+    fee : opt nat;
+    from : Account;
+    memo : opt vec nat8;
+    created_at_time : opt nat64;
+    amount : nat;
+    expected_allowance : opt nat;
+    expires_at : opt nat64;
+    spender : Account;
+};
+
+type Transaction = record {
+    burn : opt Burn;
+    kind : text;
+    mint : opt Mint;
+    approve : opt Approve;
+    timestamp : nat64;
+    transfer : opt Transfer;
+};
+
+type TransactionWithId = record {
+    id : BlockIndex;
+    transaction : Transaction;
+};
+
+type GetTransactionsResult = variant {
+    Ok: record {
+        balance: Tokens; // Current balance of the account.
+        transactions: vec TransactionWithId; // List of transactions.
+        oldest_tx_id : opt BlockIndex; // The txid of the oldest transaction the account has.
+    };
+    Err: record { message: text; };
+};
+
+type ListSubaccountsArgs = record {
+    owner: principal;
+    start: opt SubAccount;
+};
+
+type Status = record {
+    num_blocks_synced : BlockIndex;
+};
+
+service : {
```

Comment:
I wonder what is the reason that some standardized functions are called with the standard in the prefix (e.g., `icrc2_transfer_from`), while others are not.

Probably this is used when it's necessary to disambiguate new functions from priorly existing ones. 

I wonder if it would be a better convention to always have standard functions prefixed with the standard name.

Reply:
The index interface doesn't follow a standard, so the methods don't have this name spacing

Reply:
Then I wonder, should there be a note saying explicitly which canisters in the Ledger suite this standard is specifying? I now suppose it's the Ledger itself, and not the Index.

Reply:
It is specifying the index, but we take the opportunity to partially specify the index canister as well

Reply:
Added a suggestion: https://github.com/dfinity/ICRC/pull/106/files#r1858260155